src/NHibernate/Driver/OdbcDriver.cs

```diff
@@ -78,10 +78,18 @@ private void SetVariableLengthParameterSize(DbParameter dbParam, SqlType sqlType
 			{
 				switch (dbParam.DbType)
 				{
-					case DbType.AnsiString:
+					case DbType.StringFixedLength:
 					case DbType.AnsiStringFixedLength:
+						// For types that are using one character (CharType, AnsiCharType, TrueFalseType, YesNoType and EnumCharType),
+						// we have to specify the length otherwise sql function like charindex won't work as expected.
+						if (sqlType.Length == 1)
+						{
+							dbParam.Size = sqlType.Length;
+						}
```

Comment:
This may be SQL-Server specific, and maybe could create issues for people using ODBC to access another kind of database. Well, I guess no, but I am not sure.

Unless there is a way to detect the target database, maybe we need to introduce a setting for allowing to enable or disable length setting logic in the Odbc driver.

Reply:
From what I see before NH-4083 `Size` was always set and except NH-4083 I didn't found other issues related to odbc drivers and parameters. So I think we should be fine by setting the size only for char types without adding an additional setting.